queries/ocaml/highlights.scm

```diff
@@ -96,12 +96,16 @@
 
 [
   "and" "as" "assert" "begin" "class" "constraint"
-  "end" "external" "in"
-  "inherit" "initializer" "lazy" "let" "match" "method" "module"
-  "mutable" "new" "nonrec" "object" "of" "private" "rec" "sig" "struct"
-  "type" "val" "virtual" "when" "with"
+  "end" "external" "in" "inherit" "initializer"
+  "lazy" "let" "match" "method" "module"
+  "new" "object" "of" "rec" "sig" "struct"
+  "type" "val" "when" "with"
 ] @keyword
 
+[
+  "mutable" "nonrec" "private" "virtual"
```

Comment:
Reading the [ocaml manual](https://v2.ocaml.org/manual/typedecl.html), it looks like `nonrec` could be considered "type qualifier", other applied to only fields and methods, but not their _types_. Probably because how these properties are expressed. I would prefer these were kept as simple keywords

Reply:
We've discussed renaming `@type.qualifier` to something more intuitive (https://github.com/nvim-treesitter/nvim-treesitter/pull/3648#issuecomment-1279958766, https://github.com/nvim-treesitter/nvim-treesitter/pull/3648#issuecomment-1291624844).
I think it's fine to use it as is for now (even if it's not applied to types) and it can be renamed in a later PR.